ambry-router/src/main/java/com.github.ambry.router/DeleteManager.java

```diff
@@ -163,9 +163,9 @@ void close() {
       Iterator<DeleteOperation> iter = deleteOperations.iterator();
       while (iter.hasNext()) {
         DeleteOperation deleteOperation = iter.next();
+        iter.remove();
```

Comment:
Was trying to minimize the chance of completion happening twice for an operation. Have rewritten this again to ensure that completion happens not more than once. Will update the PR in some time.